frontend/packages/console-shared/src/hooks/useUserSettingsCompatibility.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { useUserSettings } from './useUserSettings';
+
+export const useUserSettingsCompatibility = <T>(
+  key: string,
+  storageKey: string,
+  defaultValue?: T,
+) => {
+  const [settings, setSettings, loaded] = useUserSettings(
```

Comment:
```suggestion
  const [settings, setSettings, loaded] = useUserSettings<T>(
```

Reply:
Have updated as below as could be string as well in case of localStorage value

```
const [settings, setSettings, loaded] = useUserSettings<string | T>(
    key,
    localStorage.getItem(storageKey) || defaultValue,
  );
```